cuqi/geometry/_geometry.py

```diff
@@ -256,6 +278,12 @@ def _create_dimension(self, dim_grid):
     @property
     def grid(self):
         return self._grid
+
+    @property
+    def par_shape(self):
+        """The shape of the parameter space."""
+        if self.fun_dim is None: return None  
+        return (self.fun_dim,) 
```

Comment:
This confuses me. Why are we using fun_dim to define the parameter dimension? To me it would make more sense to define par_shape as the original parameter and never infer that from anything else. This way the code is more explicit rather than implicit.

I guess it might mean you have to extract the parameters differently e.g:

``` Python
    @property
    def par_shape(self):
        """The shape of the parameter space."""
        if self.grid is None: return None
        return (np.size(self.grid), )
```

and

``` Python
    @property
    def par_shape(self):
        """The shape of the parameter space."""
        if self.grid is None: return None
        return np.shape(self.grid)
```

Perhaps the above approach works ok, but it is rather hard to figure out what is going on, which means at a later point we may run into errors we do not understand

Reply:
This is a great point. Now par_shape implementation  is defined independent of `fun_dim`